lib/plug/adapters/test/conn.ex

```diff
@@ -119,6 +119,16 @@ defmodule Plug.Adapters.Test.Conn do
     :ok
   end
 
+  def upgrade(%{owner: owner, ref: ref} = state, :supported = protocol, opts) do
```

Comment:
Shouldn't this be `:websocket` instead of `:supported`? Because I think folks may want to test that a given connection upgrades at certain moments?

Reply:
In fact, maybe we should allow everything and only support `:unsupported` to test the `:unsupported` cases.

Reply:
I'd intentionally used `:supported` and `:unsupported` to help illustrate that this implementation was only useful to test whether or not an upgrade is supported or not (specifically, users wouldn't get confused seeing `:websocket` and wondering why a full-fledged client upgrade didn't work). 

The test process still gets send a message in this case, which should be sufficient for users to test against. Really, all this needs to do is be able to replicate the return value for a supported and unsupported upgrade.

Reply:
But as-is it means I wouldn’t be able to test that a specific route is triggering a websocket upgrade, right? Because we would get a function clause error?

Reply:
Oh yes, I see your concern now. I suppose given that this is a test plug all bets are off with respect to upgrades (or anything else, really) actually doing anything. 

I'll update

Reply:
Done